pkg/protocols/headless/engine/page_actions.go

```diff
@@ -518,20 +549,34 @@ func (p *Page) SelectInputElement(act *Action, out ActionData) error {
 	return nil
 }
 
-// WaitLoad waits for the page to load
-func (p *Page) WaitLoad(act *Action, out ActionData) error {
-	p.page.Timeout(2 * time.Second).WaitNavigation(proto.PageLifecycleEventNameFirstMeaningfulPaint)()
-
-	// Wait for the window.onload event and also wait for the network requests
-	// to become idle for a maximum duration of 3 seconds. If the requests
-	// do not finish,
-	if err := p.page.WaitLoad(); err != nil {
-		return errors.Wrap(err, "could not wait load event")
+// WaitPageLifecycleEvent waits for specified page lifecycle event name
+func (p *Page) WaitPageLifecycleEvent(act *Action, out ActionData, event proto.PageLifecycleEventName) error {
+	fn, err := getNavigationFunc(p, act, event)
+	if err != nil {
+		return err
 	}
-	_ = p.page.WaitIdle(1 * time.Second)
+
+	fn()
+
 	return nil
 }
 
+// WaitStable waits until the page is stable
+func (p *Page) WaitStable(act *Action, out ActionData) error {
+	var err error
+	var dur time.Duration = time.Second // default 1s
+
+	argDur := act.Data["duration"]
+	if argDur != "" {
+		dur, err = time.ParseDuration(argDur)
+		if err != nil {
+			dur = time.Second
+		}
+	}
+
+	return p.page.WaitStable(dur)
```

Comment:
We should add a chained `page.Timeout` to this otherwise the stable will wait forever as the dur isn't the max wait iirc